internal/util/vault.go

```diff
@@ -65,73 +62,110 @@ Example JSON structure in the KMS config is,
 func InitVaultKMS(kmsID string, config, secrets map[string]string) (EncryptionKMS, error) {
 	var (
 		ok  bool
 		err error
 	)
+
+	vaultConfig := make(map[string]interface{})
+	keyContext := make(map[string]string)
+
 	kms := &VaultKMS{}
 	kms.EncryptionKMSID = kmsID
 
-	kms.VaultAddress, ok = config["vaultAddress"]
-	if !ok || kms.VaultAddress == "" {
+	vaultAddress, ok := config["vaultAddress"]
+	if !ok || vaultAddress == "" {
 		return nil, fmt.Errorf("missing 'vaultAddress' for vault KMS %s", kmsID)
 	}
-	kms.VaultAuthPath, ok = config["vaultAuthPath"]
-	if !ok || kms.VaultAuthPath == "" {
-		kms.VaultAuthPath = vaultDefaultAuthPath
+	vaultConfig[api.EnvVaultAddress] = vaultAddress
+
+	vaultAuthPath, ok := config["vaultAuthPath"]
+	if !ok || vaultAuthPath == "" {
+		vaultAuthPath = vaultDefaultAuthPath
 	}
-	kms.VaultRole, ok = config["vaultRole"]
-	if !ok || kms.VaultRole == "" {
-		kms.VaultRole = vaultDefaultRole
+	vaultConfig[vault.AuthMountPath], err = detectAuthMountPath(vaultAuthPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set %s in Vault config: %w", vault.AuthMountPath, err)
 	}
-	kms.VaultNamespace, ok = config["vaultNamespace"]
-	if !ok || kms.VaultNamespace == "" {
-		kms.VaultNamespace = vaultDefaultNamespace
+
+	vaultRole, ok := config["vaultRole"]
+	if !ok || vaultRole == "" {
+		vaultRole = vaultDefaultRole
 	}
-	kms.VaultPassphraseRoot, ok = config["vaultPassphraseRoot"]
-	if !ok || kms.VaultPassphraseRoot == "" {
-		kms.VaultPassphraseRoot = vaultDefaultPassphraseRoot
+	vaultConfig[vault.AuthKubernetesRole] = vaultRole
+
+	vaultNamespace, ok := config["vaultNamespace"]
+	if !ok || vaultNamespace == "" {
+		vaultNamespace = vaultDefaultNamespace
 	}
-	kms.VaultPassphrasePath, ok = config["vaultPassphrasePath"]
-	if !ok || kms.VaultPassphrasePath == "" {
-		kms.VaultPassphrasePath = vaultDefaultPassphrasePath
+	vaultConfig[api.EnvVaultNamespace] = vaultNamespace
+	keyContext[loss.KeyVaultNamespace] = vaultNamespace
+
+	// vault.VaultBackendPathKey is "secret/" by default, use vaultPassphraseRoot if configured
+	vaultPassphraseRoot, ok := config["vaultPassphraseRoot"]
+	if ok && vaultPassphraseRoot != "" {
+		// the old example did have "/v1/secret/", convert that format
+		if strings.HasPrefix(vaultPassphraseRoot, "/v1/") {
+			vaultConfig[vault.VaultBackendPathKey] = strings.TrimPrefix(vaultPassphraseRoot, "/v1/")
+		} else {
+			vaultConfig[vault.VaultBackendPathKey] = vaultPassphraseRoot
+		}
 	}
-	kms.VaultCAVerify = true
+
+	kms.vaultPassphrasePath, ok = config["vaultPassphrasePath"]
+	if !ok || kms.vaultPassphrasePath == "" {
+		kms.vaultPassphrasePath = vaultDefaultPassphrasePath
+	}
+
 	verifyCA, ok := config["vaultCAVerify"]
 	if ok {
-		kms.VaultCAVerify, err = strconv.ParseBool(verifyCA)
+		var vaultCAVerify bool
+		vaultCAVerify, err = strconv.ParseBool(verifyCA)
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse 'vaultCAVerify' for vault <%s> kms config: %s",
```

Comment:
changes not made in this PR can we use `%w` for err instead of `%s`?

Reply:
sure, will do